### PR TITLE
[MM-44329] Add a method to store.Store interface to get sql.DB object

### PR DESCRIPTION
**Pull request by noxer**
#### Summary
This PR adds two methods to the store.Store interface. `GetInternalMasterDB()` returns the `*sql.DB` for the master database. `GetInternalReplicaDBs()` returns a slice of `*sql.DB`s for the configured replica databases (or the master, when no replicas are configured). This behavior is implemented to emulate the functionality of `GetMasterX()` and `GetReplicaX()`.
This PR also updates the mocks to include the new methods.

As an internal change, this adds a method `"sqlStore".Store.hasLicense()` which returns `true` iff the license is not `nil`. Putting it into it's own method allows for simpler (and more efficient) code.

#### Ticket Link
[MM-44329](https://mattermost.atlassian.net/browse/MM-44329)

```release-note
NONE
```

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -66,6 +67,8 @@ type Store interface {
 	GetDBSchemaVersion() (int, error)
 	GetAppliedMigrations() ([]model.AppliedMigration, error)
 	GetDbVersion(numerical bool) (string, error)
+	GetInternalMasterDB() *sql.DB
+	GetInternalReplicaDBs() []*sql.DB
```
Comment (agnivade) [inline store/store.go:71]: Maybe worth adding a comment here to explain that these are added for other products in the multi-product architecture to get raw access to the DB. Just for someone wondering why these are exposed.

**Review by agnivade — APPROVED** [review]
Nicely done!

File: store/store.go (hunk 1 above)
Comment (noxer) [reply store/store.go:71]: Yeah, that makes sense.

Comment (noxer) [reply store/store.go]: I added comments to both methods in the interface.

**Review by isacikgoz — APPROVED** [review]
LGTM, well done!

**Pull request merged**